assets/src/edit-story/app/media/useMediaReducer/reducer.js

```diff
@@ -53,36 +60,42 @@ function reducer(state, { type, payload }) {
     case types.FETCH_MEDIA_ERROR: {
       return {
         ...state,
-        media: [],
         isMediaLoaded: true,
         isMediaLoading: false,
       };
     }
 
     case types.RESET_FILTERS: {
-      return {
-        ...state,
-        searchTerm: '',
-        mediaType: '',
-      };
+      return { ...INITIAL_STATE };
     }
 
     case types.SET_SEARCH_TERM: {
       const { searchTerm } = payload;
+      if (searchTerm === state.searchTerm) return state;
       return {
-        ...state,
+        ...INITIAL_STATE,
+        mediaType: state.mediaType,
         searchTerm,
       };
     }
 
     case types.SET_MEDIA_TYPE: {
       const { mediaType } = payload;
+      if (mediaType === state.mediaType) return state;
       return {
-        ...state,
+        ...INITIAL_STATE,
```

Comment:
Probably the search term should prevail when switching between media types?

Reply:
Correct, in this old UI, it's not so obvious, in Figma it's clear `select` (Filters can be applied on top of a search).
Done in https://github.com/google/web-stories-wp/pull/562/commits/96e2fb4ba24a03d802a4f7a877506afed9083506